CrazyCanvas/Source/ECS/Systems/GUI/HUDSystem.cpp

```diff
@@ -281,6 +277,32 @@ bool HUDSystem::OnMatchCountdownEvent(const MatchCountdownEvent& event)
 	return false;
 }
 
+bool HUDSystem::OnProjectedEntityAdded(LambdaEngine::Entity projectedEntity)
+{
+	ECSCore* pECS = ECSCore::GetInstance();
+	const ComponentArray<ProjectedGUIComponent>* pProjectedGUIComponents = pECS->GetComponentArray<ProjectedGUIComponent>();
+	const ProjectedGUIComponent& projectedGUIComponent = pProjectedGUIComponents->GetConstData(projectedEntity);
+
+
+	if (projectedGUIComponent.GUIType == IndicatorTypeGUI::FLAG_INDICATOR)
+	{
+		const ComponentArray<TeamComponent>* pTeamComponents = pECS->GetComponentArray<TeamComponent>();
+		const TeamComponent& teamComponent = pTeamComponents->GetConstData(projectedEntity);
+		m_HUDGUI->CreateProjectedGUIElement(projectedEntity, m_LocalTeamIndex, teamComponent.TeamIndex);
+	}
+	else
+	{
+		m_HUDGUI->CreateProjectedGUIElement(projectedEntity, m_LocalTeamIndex);
+	}
+
+	return false;
+}
+
+bool HUDSystem::RemoveProjectedEntity(LambdaEngine::Entity projectedEntity)
+{
+	return false;
+}
```

Comment:
Should this be empty? I saw the callback was commented

Reply:
Right now it's just an empty definition without a callback attached. I'm not sure if this will ever be needed. Maybe I should just remove it entirely? It was a "just in case" implementation